modules/secure-source-manager-instance/iam.tf

```diff
@@ -33,29 +33,31 @@ locals {
 
 resource "google_secure_source_manager_instance_iam_binding" "authoritative" {
   for_each    = var.iam
-  project     = google_secure_source_manager_instance.instance.project
-  location    = google_secure_source_manager_instance.instance.location
-  instance_id = google_secure_source_manager_instance.instance.instance_id
+  project     = try(google_secure_source_manager_instance.instance[0].project, var.project_id)
+  location    = try(google_secure_source_manager_instance.instance[0].location, var.location)
+  instance_id = try(google_secure_source_manager_instance.instance[0].instance_id, var.instance_id)
```

Comment:
nit: is there any way to remove the try() calls. Maybe having a local for each using a ternary? Something like
```terraform
locals {
  location = var.instance_create ? google_secure_source_manager_instance.instance[0].location : var.location
}
```

with this you can also remove the depends_on on the other resources